pilot/pkg/serviceregistry/kube/controller/namespacecontroller.go

```diff
@@ -134,7 +96,36 @@ func (nc *NamespaceController) Run(stopCh <-chan struct{}) {
 		log.Error("Failed to sync namespace controller cache")
 	}
 	log.Infof("Namespace controller started")
-	go nc.queue.Run(stopCh)
+
+	go wait.Until(nc.runWorker, time.Second, stopCh)
```

Comment:
Why do we need to call runWorker many times? it exists after the queue is shutdown, so re-running it seems wrong. The comment also says 'It returns false when it's time to quit.'

Reply:
To clarify - I know k8s does this, I have just never understood why...

Reply:
I think the `wait.Until` can be used to catch unexpected exceptions to avoid the goroutine being crashed (Even though it seems no exception will be thrown in the `runWorker`).
And the `worker` will be stopped only when `stopCh` is triggered, which will also cause the `queue` to be `shutDown`.